File: Java (Task 3)/src/main/java/com/example/custombroadcastreceiverapp/MainActivity.java
package com.example.custombroadcastreceiverapp;  // Replace 'com.example.myapp' with your actual package name

import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;

public class MainActivity extends AppCompatActivity {
    private MyCustomBroadcastReceiver myReceiver;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Initialize and set up the custom BroadcastReceiver
        myReceiver = new MyCustomBroadcastReceiver();

        Button sendBroadcastButton = findViewById(R.id.sendBroadcastButton);
        sendBroadcastButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Create and send an intent with the custom action
                Intent intent = new Intent("com.example.CUSTOM_ACTION");
                LocalBroadcastManager.getInstance(MainActivity.this).sendBroadcast(intent);
            }
        });
    }

    @Override
    protected void onStart() {
        super.onStart();
        // Register the receiver with the intent filter
        LocalBroadcastManager.getInstance(this).registerReceiver(myReceiver, new IntentFilter("com.example.CUSTOM_ACTION"));
    }

    @Override
    protected void onStop() {
        super.onStop();
        // Unregister the receiver to prevent memory leaks
        LocalBroadcastManager.getInstance(this).unregisterReceiver(myReceiver);
    }
}

File: Kotlin (Task3)/src/main/java/com/example/custombroadcastreceiverapp/MainActivity.kt
package com.example.custombroadcastreceiverapp // Ensure this matches your package name

import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.localbroadcastmanager.content.LocalBroadcastManager

class MainActivity : AppCompatActivity() {
    private lateinit var sendBroadcastButton: Button // Button to send broadcast
    private lateinit var myReceiver: MyCustomBroadcastReceiver // Custom Broadcast Receiver instance

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main) // Set the layout resource

        // Initialize the Button using findViewById
        sendBroadcastButton = findViewById(R.id.sendBroadcastButton)

        // Initialize the BroadcastReceiver
        myReceiver = MyCustomBroadcastReceiver()

        // Set up button click listener
        sendBroadcastButton.setOnClickListener {
            val intent = Intent("com.example.CUSTOM_ACTION")
            LocalBroadcastManager.getInstance(this).sendBroadcast(intent)
        }
    }

    override fun onStart() {
        super.onStart()
        // Register the BroadcastReceiver
        LocalBroadcastManager.getInstance(this).registerReceiver(
            myReceiver,
            IntentFilter("com.example.CUSTOM_ACTION")
        )
    }

    override fun onStop() {
        super.onStop()
        // Unregister the BroadcastReceiver
        LocalBroadcastManager.getInstance(this).unregisterReceiver(myReceiver)
    }
}
